src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,170 @@
 package core.basesyntax;
 
+//import jdk.internal.util.ArraysSupport;
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+
 public class ArrayList<T> implements List<T> {
+
+
+    private static final int def_capacity = 10;
+    private static final Object[] empty_arraydata = {};
+
+
+    private T[] array = (T[]) new Object[def_capacity];
```

Comment:
it is desirable to set the array through the constructor with the annotation @SuppressWarnings("unchecked") in this case:

@SuppressWarnings("unchecked")
public ArrayList() {     
    this.array = (T[]) new Object[def_capacity];
}